### PR TITLE
ApplyMixin

### PR DESCRIPTION
**Pull request by sergiodxa**
I was using it and find a case we don’t see before.

If the mixin defines some **contextTypes** and the class also do it then the mixin contextTypes don’t apply.

I added a condition, that if the mixin key is **contextTypes** then there are merged into one single contextType.

Also this will work in the future for other props that can be merged and are objects (maybe propTypes?).

### CODE REVIEW

**Comment by sergiodxa** [comment]
An example of use case:

``` javascript
import React from 'react';
import { IntlMixin } from 'react-intl';

@applyMixin(IntlMixin)
class Component extends React.Component {
  static contextTypes = {
    keyName: React.PropTypes.object,
  };
  ...
}
```

In this example I don't have the **contextTypes** that **IntlMixin** defines, and that give me errors, with this change IntlMixin contextTypes and my own contextTypes will merge.

File: src/applyMixin.js [hunk 1]
```diff
@@ -1,7 +1,49 @@
...
 function applyMixin(target, mixin) {
   // apply mixin to the target class
   // and return the mixed class
   return Object.keys(mixin).reduce((targetClass, key) => {
+    // if the key is contextTypes
+    if (key === 'contextTypes') {
+      // merge the target and mixin contextTypes
+      const value = Object.assign({}, mixin[key], targetClass[key]);
+      // apply it
+      targetClass[key] = value;
```
Comment (goncalvesjoao) [inline src/applyMixin.js:16]: checkout if this change still does what you want

``` javascript
targetClass[key] = { ...mixin[key], ...targetClass[key] };
```

When we run **npm run build** babel seems to cover our needs and we don't have to declare Object.assign ourselves.

In the mean time I'll be writing some tests to cover what you said.

**Comment by sergiodxa** [comment]
I used your solution, that works fine :+1:

**Comment by goncalvesjoao** [comment]
Thanks again Sergio!
Just published version 1.3.1 to npm.

**Pull request merged**